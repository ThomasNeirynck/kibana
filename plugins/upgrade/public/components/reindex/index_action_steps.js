import React from 'react';

import { IndexActionStep } from './index_action_step';


export function IndexActionSteps({ steps, action, indexName }) {
  return (
    <div>
      { steps.map((step, stepIndex) => (
        <IndexActionStep
          key={ stepIndex }
          step={ step }
          action={ action }
          indexName={ indexName }
        />
      )) }
    </div>
  );
}

IndexActionSteps.propTypes = {
  steps: React.PropTypes.array,
  action: React.PropTypes.string,
  indexName: React.PropTypes.string,
};

IndexActionSteps.defaultProps = {
  steps: [],
  action: React.PropTypes.string,
  indexName: React.PropTypes.string,
};
