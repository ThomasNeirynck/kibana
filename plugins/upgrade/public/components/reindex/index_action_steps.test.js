import { shallow } from 'enzyme';
import React from 'react';

import { IndexActionSteps } from './index_action_steps';

jest.mock('ui/chrome', () => {}, { virtual: true });
jest.mock('ui/notify/notifier', () => ({
  Notifier: function Notifier() {},
}), { virtual: true });


describe('IndexActionSteps', () => {
  test('renders an IndexActionStep for each of the steps', () => {
    const component = (
      <IndexActionSteps
        steps={ [
          { name: 'STEP 1' },
          { name: 'STEP 2' },
          { name: 'STEP 3' },
        ] }
        action="ACTION"
        indexName="INDEX_NAME"
      />
    );

    expect(shallow(component)).toMatchSnapshot();
  });
});
