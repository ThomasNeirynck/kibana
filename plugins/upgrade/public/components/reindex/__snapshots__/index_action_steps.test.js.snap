// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexActionSteps renders an IndexActionStep for each of the steps 1`] = `
<div>
  <IndexActionStep
    action="ACTION"
    indexName="INDEX_NAME"
    step={
      Object {
        "name": "STEP 1",
      }
    }
    steps={Object {}}
  />
  <IndexActionStep
    action="ACTION"
    indexName="INDEX_NAME"
    step={
      Object {
        "name": "STEP 2",
      }
    }
    steps={Object {}}
  />
  <IndexActionStep
    action="ACTION"
    indexName="INDEX_NAME"
    step={
      Object {
        "name": "STEP 3",
      }
    }
    steps={Object {}}
  />
</div>
`;
